pages/api/habits/[id].js

```diff
@@ -0,0 +1,56 @@
+import dbConnect from "../../../lib/dbConnect";
+import Habit from "../../../models/Habit";
+
+export default async function handler(req, res) {
+  const {
+    query: { id },
+    method,
+  } = req;
+
+  await dbConnect();
+
+  switch (method) {
+    case "GET" /* Get a model by its ID */:
+      try {
+        const habit = await Habit.findById(id);
+        if (!habit) {
```

Comment:
i think this if statement is already handled by line 21. You should check if this is necessary (i am not sure)